### PR TITLE
[ParticleMechanics] Porting to Amatrix and some minor cleanup

### PR DESCRIPTION
**Pull request by bodhinandach**
Mainly porting ParticleMechanicsApplication according to PR #2765.

So far this compiles with the current ublas.

Howeer, there could be some missing updates but I cannot check as I need to wait for the SolidMechanicsApplication and ConstitutiveModelsApplication to port as well.
@josep-m-carbonell could you please notify me once they are updated? Thanks.

### CODE REVIEW

File: applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp [hunk 1]
```diff
@@ -209,99 +209,99 @@ bool MCPlasticFlowRule::CalculateConsistencyCondition(RadialReturnVariables& rRe
...
-    N2(0) = R_p(1)*R_p2(2) - R_p(2)*R_p2(1);
-    N2(1) = R_p(2)*R_p2(0) - R_p(0)*R_p2(2);
-    N2(2) = R_p(0)*R_p2(1) - R_p(1)*R_p2(0);
+    N2[0] = R_p[1]*R_p2[2] - R_p[2]*R_p2[1];
+    N2[1] = R_p[2]*R_p2[0] - R_p[0]*R_p2[2];
+    N2[2] = R_p[0]*R_p2[1] - R_p[1]*R_p2[0];
 
-    double num1 = N2(0) * sigma_P_apex(0) + N2(1) * sigma_P_apex(1) + N2(2) * sigma_P_apex(2);
-    double den_1 = N2(0) + N2(1) + friction_coefficient * N2(2);
+    double num1 = N2[0] * sigma_P_apex[0] + N2[1] * sigma_P_apex[1] + N2[2] * sigma_P_apex[2];
```
Comment (loumalouomega) [inline applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp]: const

File: applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp [hunk 2]
```diff
@@ -209,99 +209,99 @@ bool MCPlasticFlowRule::CalculateConsistencyCondition(RadialReturnVariables& rRe
...
+    double den_1 = N2[0] + N2[1] + friction_coefficient * N2[2];
     const double t1 = num1 / den_1 ;
 
     // Secondary surface in region III a = [k -1 0], b  = [m -1 0] -- needed to calculate t2
     double den = friction_coefficient * (D(0,0) * dilatancy_coefficient - D(0,1)) - D(1,0) * dilatancy_coefficient + D(1,1);
```
Comment (loumalouomega) [inline applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp]: const, check non zero

File: applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp [hunk 3]
```diff
@@ -399,9 +399,9 @@ void MCPlasticFlowRule::CalculatePrincipalStressTrial(const RadialReturnVariable
 
     // Calculate the elastic matrix
     Matrix ElasticMatrix = ZeroMatrix(3,3);
-    const double& young_modulus     = mpYieldCriterion->GetHardeningLaw().GetProperties()[YOUNG_MODULUS];
-    const double& poisson_ratio        = mpYieldCriterion->GetHardeningLaw().GetProperties()[POISSON_RATIO];
-    const double diagonal   = young_modulus/(1.0+poisson_ratio)/(1.0-2.0*poisson_ratio) * (1.0-poisson_ratio);
+    const double& young_modulus = mpYieldCriterion->GetHardeningLaw().GetProperties()[YOUNG_MODULUS];
```
Comment (loumalouomega) [inline applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp]: const double& == const double. The optimizer corrects this, but please change

**Review by loumalouomega — DISMISSED** [review]
look my comments

**Review by loumalouomega — APPROVED** [review]

**Comment by bodhinandach** [comment]
thanks Vicente @loumalouomega !

**Comment by josep-m-carbonell** [comment]
Hi @bodhinandach  
      There is still some methods that does not exist in Amatrix, so they can not be ported. I am waiting for a solution in order to complete the porting of the SolidMechanicsApplication.

**Comment by bodhinandach** [comment]
@josep-m-carbonell I understood. No need to rush. I just want to check the ParticleMechanics once you ported, so will be awesome if you can notify me. Thanks 😃

**Pull request merged**